evm-contracts/src/v0.6/dev/FluxAggregator.sol

```diff
@@ -674,12 +674,16 @@ contract FluxAggregator is AggregatorInterface, Owned {
   function roundState(address _oracle)
     external
     view
-    returns (uint32 _reportableRoundId, bool _eligibleToSubmit, int256 _latestRoundAnswer)
+    returns (uint32 _reportableRoundId, bool _eligibleToSubmit, int256 _latestRoundAnswer, uint64 _timesOutAt)
   {
-    uint32 roundId = rounds[reportingRoundId].details.answers.length == rounds[reportingRoundId].details.maxAnswers || timedOut(reportingRoundId)
-      ? reportingRoundId.add(1)
-      : reportingRoundId;
-    return (roundId, eligibleToSubmit(_oracle), rounds[latestRoundId].answer);
+    bool finishedOrTimedOut = rounds[reportingRoundId].details.answers.length >= rounds[reportingRoundId].details.maxAnswers || timedOut(reportingRoundId);
+    uint32 reportableRoundId = finishedOrTimedOut ? reportingRoundId.add(1) : reportingRoundId;
```

Comment:
Not sure I really agree with the logic behind `reportableRoundId` both here and in `eligibleToSubmit`...

Let's say we have a fluxAgg with `minAnswers = 1` and `maxAnswers = 2`, and 2 oracles...

After oracle 1 submits an answer to round 1, and we call `roundState(oracle2.address)`, we would see reportableRoundId = 2... but here oracle 2 is actually eligible to report for either round 1 or round 2. In an effort to maximize the number of oracles reporting per round, I think we want to _round down_ here and instead return reportableRoundId = 1.

Reply:
Yeah, this is exactly why I'm using `maxAnswers` instead of `minAnswers` in the `roundState()` method to determine which round is reportable.

However, you're definitely correct about `eligibleToSubmit()` -- I forgot to update that one.  Will fix.

Reply:
ahh okay that actually makes sense. and we're changing `reportableRoundId` in `eligibleToSubmit()` ??